### PR TITLE
rpc/rm_92173_proposal_open_contract_response

### PR DESCRIPTION
**Pull request by tanvir-deriv**

### CODE REVIEW

File: lib/Finance/Calendar.pm [hunk 1]
```diff
@@ -789,12 +789,18 @@ sub next_open_at {
...
         my $trading_breaks = $self->trading_breaks($exchange, $date);
+        
         foreach my $break ($trading_breaks->@*) {
             my ($close, $open) = $break->@*;
             if ($date->is_after($close) and $date->is_before($open)) {
                 return $open;
             }
         }
+
+        unless (scalar $trading_breaks == 0) {
```
[the remark this answers is not in the document]
Comment (tanvir-deriv) [reply lib/Finance/Calendar.pm:800]: `$trading_breaks` is getting an array with `close` and `open` times. In the case of no trading breaks for an exchange, it is getting an empty array. I can update that.

[the remark this answers is not in the document]
Comment (tanvir-deriv) [reply lib/Finance/Calendar.pm:800]: > Also, would it be possible to add a test?

Let me check

**Comment by qinfeng-deriv** [comment]
Please add how this solution works in card description also

File: lib/Finance/Calendar.pm (hunk 1 above)
[the remark this answers is not in the document]
Comment (tanvir-deriv) [reply lib/Finance/Calendar.pm:800]: Yes. Updated. Please take a look.

**Comment by tanvir-deriv** [comment]
> Please add how this solution works in card description also

Added. Please check and let me know.

File: lib/Finance/Calendar.pm (hunk 1 above)
[the remark this answers is not in the document]
Comment (tanvir-deriv) [reply lib/Finance/Calendar.pm]: > Also, would it be possible to add a test?

I have added tests for next_open_at. Please check

**Review by qinfeng-deriv — APPROVED** [review]

**Pull request merged**